source/extensions/filters/http/abac/abac_filter.cc

```diff
@@ -0,0 +1,171 @@
+#include "extensions/filters/http/abac/abac_filter.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ABACFilter {
+
+using CelValue = google::api::expr::runtime::CelValue;
+using CelMap = google::api::expr::runtime::CelMap;
+using CelList = google::api::expr::runtime::CelList;
+
+namespace {
+
+constexpr absl::string_view kMetadata = "metadata";
+constexpr absl::string_view kConnection = "connection";
+constexpr absl::string_view kRequestedServerName = "requested_server_name";
+constexpr absl::string_view kLocalAddress = "local_address";
+constexpr absl::string_view kRemoteAddress = "remote_address";
+constexpr absl::string_view kCertificatePresented = "certificate_presented";
+constexpr absl::string_view kSubjectLocalCertificate = "subject_local_certificate";
+constexpr absl::string_view kIssuerPeerCertificate = "issuer_peer_certificate";
+constexpr absl::string_view kSubjectPeerCertificate = "subject_peer_certificate";
+constexpr absl::string_view kTlsVersion = "tls_version";
+constexpr absl::string_view kRequest = "request";
+constexpr absl::string_view kPath = "path";
+constexpr absl::string_view kHost = "host";
+constexpr absl::string_view kScheme = "scheme";
+constexpr absl::string_view kMethod = "method";
+constexpr absl::string_view kReferer = "referer";
+constexpr absl::string_view kHeaders = "headers";
+constexpr absl::string_view kTime = "time";
+
+class ConnectionWrapper : public CelMap {
+public:
+  ConnectionWrapper(const StreamInfo::StreamInfo& info) : info_(info) {}
+
+  absl::optional<CelValue> operator[](CelValue key) const override {
+    if (!key.IsString()) {
+      return {};
+    }
+    auto value = key.StringOrDie().value();
+    if (value == kRequestedServerName) {
```

Comment:
can those be something like flat_hash_map<string_view, std::function>? I think we do similar in access_log_formatter.

Reply:
I'll need to measure it. std::function are surprisingly expensive in terms of unexpected allocations. The way it's done here is trivially allocation-free.

Reply:
You don't have to create a map every time, have a static map like: flat_hash_map<string_view, std::function<CelValue(const StreamInfo&)>>, so you just call them every time, not allocating.

Reply:
I'm not sure that's more readable. Anyways, I _need_ to measure before making the change.

Reply:
Lizan, is this what you had in mind https://github.com/kyessenov/envoy/blob/abac_filter_feedback/test/extensions/filters/common/expr/context_speed_test.cc#L34?

Reply:
yep, thanks for testing that out.

Reply:
After adding all functions to the flat_hash_map, here is the performance comparison:
```
2019-08-05 14:33:56
Running bazel-bin/test/extensions/filters/common/expr/context_speed_test
Run on (16 X 2200 MHz CPU s)
CPU Caches:
  L1 Data 32K (x8)
  L1 Instruction 32K (x8)
  L2 Unified 256K (x8)
  L3 Unified 56320K (x1)
Load Average: 1.14, 1.13, 1.19
---------------------------------------------------------
Benchmark               Time             CPU   Iterations
---------------------------------------------------------
BM_Context           17.6 ns         17.6 ns     39715762
BM_TestContext       17.2 ns         17.2 ns     42524637
```

So it's roughly the same (I couldn't trivially wire a memory allocation profiler), and it comes down to readability and code style.